### PR TITLE
Specify user and group in systemd unit file

### PR DESCRIPTION
**Pull request by Jules-Bertholet**
Fix CI warning

## Problem
- User and group were missing

## Solution
- Added them

## PR Status
- [x] Code finished.
- [ ] Tested with Package_check.
- [x] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Package_check results
---
* An automatic package_check will be launch at https://ci-apps-dev.yunohost.org/, when you add a specific comment to your Pull Request: "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!"*

### CODE REVIEW

**Comment by Jules-Bertholet** [comment]
!testme

File: conf/systemd.service [hunk 1]
```diff
@@ -6,10 +6,12 @@ After=syslog.target network.target
 Environment="LC_ALL=C.UTF-8"
 Environment="LANG=C.UTF-8"
 Environment="PATH=__NODE_PATH__:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+USER=__APP__
+GROUP=__APP__
```
Comment (ericgaspar) [inline conf/systemd.service:10]:
```suggestion
User=__APP__
Group=__APP__
```

**Review by ericgaspar — CHANGES_REQUESTED** [review]
The CI still complains about `User:` not being set... Also I am not sure how systemd reacts to capitalization...

**Comment by ericgaspar** [comment]
!testme

**Comment by Jules-Bertholet** [comment]
CI passed, this is ready to merge

**Comment by ericgaspar** [comment]
After installing amd login I get  `500 : Internal Server Error Error in Authenticator.pre_spawn_start: SubprocessError Exception occurred in preexec_fn.` error. I seems related to this issue: https://github.com/jupyterhub/jupyterhub/issues/1527

**Comment by Jules-Bertholet** [comment]
!testme

**Comment by kay0u** [comment]
Thank you for this very interesting PR!
I know it's very tempting to refactor scripts to avoid copying the same code everywhere, but we chose not to do that on the `example_ynh` template for some reason, and we want to keep the packages as close to that template as possible. Can you avoid doing this, please?

**Comment by Jules-Bertholet** [comment]
> I know it's very tempting to refactor scripts to avoid copying the same code everywhere, but we chose not to do that on the `example_ynh` template for some reason, and we want to keep the packages as close to that template as possible. Can you avoid doing this, please?

It was so tempting that I proposed https://github.com/YunoHost/example_ynh/pull/164.

I've tested everything manually, this should be ready for review now

File: scripts/install [hunk 2]
```diff
@@ -143,8 +149,27 @@ ynh_add_config --template="../conf/jupyter_notebook_config.py" --destination="$f
...
+chmod -R g=u,g-w,o-rwx "$final_path"
+setfacl -nR -m g:$app.main:rx -m d:g:$app.main:rx "$final_path/.venv"
+setfacl -n -m g:$app.main:x "$final_path"
+chown root:root "/etc/sudoers.d/$app-sudoers"
+chmod 440 "/etc/sudoers.d/$app-sudoers"
+
+#=================================================
+# BUILD USER LABS
+#=================================================
+ynh_script_progression --message="Building Jupyter lab for each user..." --weight=10
```
Comment (ericgaspar) [inline scripts/install:163]:
```suggestion
ynh_script_progression --message="Building JupyterLab for each user..." --weight=10
```

File: scripts/upgrade [hunk 3]
```diff
@@ -157,8 +165,28 @@ ynh_add_systemd_config
...
+setfacl -nR -m g:$app.main:rx -m d:g:$app.main:rx "$final_path/.venv"
+setfacl -n -m g:$app.main:x "$final_path"
+chown root:root "/etc/sudoers.d/$app-sudoers"
+chmod 440 "/etc/sudoers.d/$app-sudoers"
+
+
+#=================================================
+# BUILD USER LABS
+#=================================================
+ynh_script_progression --message="Building Jupyter lab for each user..." --weight=10
```
Comment (ericgaspar) [inline scripts/upgrade:180]:
```suggestion
ynh_script_progression --message="Building JupyterLab for each user..." --weight=10
```

**Review by ericgaspar — CHANGES_REQUESTED** [review]
minor typos

**Review by ericgaspar — APPROVED** [review]

**Pull request merged**